### PR TITLE
Minimal changes for an offline mode

### PR DESCRIPTION
**Pull request by jimchamp**
Added new config `offline_mode`, which, when true, mocks calls to Internet Archive. Additionally, the homepage stats component will not render while in offline mode, as these stats are being fetched from IA.

These changes speed up and remove errors from the following pages:
- The homepage
- Search pages
- Book pages
- My Books pages

### CODE REVIEW

File: openlibrary/plugins/openlibrary/code.py [hunk 1]
```diff
@@ -1241,7 +1245,7 @@ def GET(self):
             query = i.get('data', '')
             data = fulltext_search(query)
             hits = data.get('hits', [])
-            if not hits['hits']:
+            if not hits:
```
Comment (jimchamp) [inline openlibrary/plugins/openlibrary/code.py:1248]: This change seems to address an untracked bug, and can possibly be made on `master` as well.

File: openlibrary/templates/home/index.html [hunk 2]
```diff
@@ -40,7 +40,8 @@
 
   $:render_template("home/categories", test=test)
 
-  $:render_template("home/stats", stats)
+  $if not hide_stats:
+    $:render_template("home/stats", stats)
```
Comment (jimchamp) [inline openlibrary/templates/home/index.html:44]: Several elements in [`home/stats`](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/home/stats.html) source data from IA.  It seems cleaner to just remove the component while in offline-mode.

File: openlibrary/plugins/upstream/utils.py [hunk 3]
```diff
@@ -1456,7 +1456,7 @@ def get_donation_include() -> str:
     """
         % script_src
     )
-    return html
+    return html if config.get("offline_mode", False) else ""
```
Comment (jimchamp) [inline openlibrary/plugins/upstream/utils.py:1459]: Returns empty string for donations scripts when in offline-mode.

File: openlibrary/core/sponsorships.py [hunk 4]
```diff
@@ -61,6 +62,8 @@ def get_sponsored_editions(user, page: int = 1) -> list:
     :param user user: infogami user
     :return: list of archive.org editions sponsored by user
     """
+    if config.get('offline_mode', False):
+        return []
```
Comment (jimchamp) [inline openlibrary/core/sponsorships.py:66]: Returns empty list of sponsorship data while in offline-mode.  This data is used to populate the sidebar in "My Books" views, even for patrons who did not participate in the program while it was active.

**Review by scottbarnes — APPROVED** [review]
With changes to address editing of works and editions, this looks good to me. I also tested importing locally. My vote is to go with this and we can update the PR if necessary.

**Comment by jimchamp** [comment]
Thanks @scottbarnes!  Have applied, tested, and pushed your changes, and will now merge.

**Pull request merged**